oda_api/api.py

```diff
@@ -191,13 +188,20 @@ def __init__(self,
             if host.startswith('http'):
                 self.url = host
             else:
-                if protocol == 'http':
-                    self.url = "http://" + host
-                elif protocol == 'https':
-                    self.url = "https://" + host
-                else:
+                if protocol != 'http' and protocol != 'https':
                     raise UserError('protocol must be either http or https')
+                else:
+                    self.url = protocol + "://" + host
         else:
+            if url is None:
+                url = "https://www.astro.unige.ch/mmoda/dispatch-data"
+            else:
+                if not url.startswith('http'):
+                    if protocol != 'http' and protocol != 'https':
+                        raise UserError('protocol must be either http or https')
+                    else:
+                        url = protocol + "://" + url
```

Comment:
this is very incorrect, URL already includes "protocol" (actually, scheme, which represents protocol in subtle way), by [definition](https://datatracker.ietf.org/doc/html/rfc3986). If `url` is provided,  every other connection parameter (port, host, protocol) should be discarded. And later obsolete.

The fact that this division of URL was even introduced is a result of misunderstanding of what URL is.